arch/TriCore/TriCoreMapping.c

```diff
@@ -130,4 +121,120 @@ const char *TriCore_group_name(csh handle, unsigned int id)
+typedef struct insn_op {
+	uint8_t /* cs_op_type */ type; ///< Operand type (e.g.: reg, imm, mem)
+	uint8_t /* cs_ac_type */ access; ///< The access type (read, write)
+	uint8_t /* cs_data_type */
+		dtypes[10]; ///< List of op types. Terminated by CS_DATA_TYPE_LAST
+} insn_op;
+
+///< Operands of an instruction.
+typedef struct {
+	insn_op ops[16]; ///< NULL terminated array of operands.
+} insn_ops;
+
+const insn_ops insn_operands[] = {
+#include "TriCoreGenCSMappingInsnOp.inc"
+};
+
+static inline cs_op_type TriCore_get_op_type(MCInst *MI, unsigned OpNum)
+{
+	assert(MI->Opcode < ARR_SIZE(insn_operands));
+	assert(OpNum < ARR_SIZE(insn_operands[MI->Opcode].ops));
+
+	return insn_operands[MI->Opcode].ops[OpNum].type;
+}
+
+static inline cs_ac_type TriCore_op_access(MCInst *MI, unsigned OpNum)
+{
+	assert(MI->Opcode < ARR_SIZE(insn_operands));
+	assert(OpNum < ARR_SIZE(insn_operands[MI->Opcode].ops));
+
+	cs_ac_type access = insn_operands[MI->Opcode].ops[OpNum].access;
+	if (MCInst_opIsTied(MI, OpNum) || MCInst_opIsTying(MI, OpNum))
+		access |= (access == CS_AC_READ) ? CS_AC_WRITE : CS_AC_READ;
+	return access;
+}
 #endif
+
+void TriCore_set_access(MCInst *MI)
+{
+#ifndef CAPSTONE_DIET
+	if (!(MI->csh->detail == CS_OPT_ON && MI->flat_insn->detail))
+		return;
+
+	assert(MI->Opcode < ARR_SIZE(insn_operands));
+
+	cs_detail *detail = MI->flat_insn->detail;
+	cs_tricore *tc = &(detail->tricore);
+	for (int i = 0; i < tc->op_count; ++i) {
+		cs_ac_type ac = TriCore_op_access(MI, i);
+		cs_tricore_op *op = &tc->operands[i];
+		op->access = ac;
+		cs_op_type op_type = TriCore_get_op_type(MI, i);
+		if (op_type != CS_OP_REG) {
+			continue;
+		}
+		if (ac & CS_AC_READ) {
+			detail->regs_read[detail->regs_read_count++] = op->reg;
+		}
+		if (ac & CS_AC_WRITE) {
+			detail->regs_write[detail->regs_write_count++] =
+				op->reg;
+		}
+	}
+#endif
+}
+
+void TriCore_reg_access(const cs_insn *insn, cs_regs regs_read,
+			uint8_t *regs_read_count, cs_regs regs_write,
+			uint8_t *regs_write_count)
+{
+#ifndef CAPSTONE_DIET
+	uint8_t read_count, write_count;
+	cs_detail *detail = insn->detail;
+	read_count = detail->regs_read_count;
+	write_count = detail->regs_write_count;
+
+	// implicit registers
+	memcpy(regs_read, detail->regs_read,
+	       read_count * sizeof(detail->regs_read[0]));
+	memcpy(regs_write, detail->regs_write,
+	       write_count * sizeof(detail->regs_write[0]));
+
+	// explicit registers
+	cs_tricore *tc = &detail->tricore;
+	for (uint8_t i = 0; i < tc->op_count; i++) {
+		cs_tricore_op *op = &(tc->operands[i]);
+		switch ((int)op->type) {
+		case TRICORE_OP_REG:
+			if ((op->access & CS_AC_READ) &&
+			    !arr_exist(regs_read, read_count, op->reg)) {
+				regs_read[read_count] = (uint16_t)op->reg;
+				read_count++;
+			}
+			if ((op->access & CS_AC_WRITE) &&
+			    !arr_exist(regs_write, write_count, op->reg)) {
+				regs_write[write_count] = (uint16_t)op->reg;
+				write_count++;
+			}
+			break;
+		case TRICORE_OP_MEM:
```

Comment:
You have not defined any memory operands in TriCore. But if you do in the future, you must check it via `op->type & TRICORE_OP_MEM`. Because `CS_OP_MEM` is an flag. Not an operand on its own.
Please pull in https://github.com/capstone-engine/capstone/pull/2013/commits/22dbffaee5a2add0178e695d2cb44363f349148d and update the `TRICORE_OP_` enum just as I did [there for ARM](https://github.com/capstone-engine/capstone/blob/22dbffaee5a2add0178e695d2cb44363f349148d/include/capstone/arm.h#L399-L412)

Reply:
I made some changes, but the `case TRICORE_OP_MEM:` part I want to wait until it is needed in the future.